buildSrc/src/integTest/groovy/org/elasticsearch/gradle/YamlRestCompatTestPluginFuncTest.groovy

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleFuncTest
+import org.gradle.testkit.runner.TaskOutcome
+import spock.lang.IgnoreIf
+
+class YamlRestCompatTestPluginFuncTest extends AbstractGradleFuncTest {
+
+    def "yamlRestCompatTest does nothing when there are no tests"() {
+        given:
+
+        addSubProject(":distribution:bwc:minor") <<  """
+        configurations { checkout }
+        artifacts {
+            checkout(new File(projectDir, "checkoutDir"))
+        }
+        """
+
+        buildFile << """
+        plugins {
+          id 'elasticsearch.yaml-rest-compat-test'
+        }
+        """
+
+        when:
+        def result = gradleRunner("yamlRestCompatTest", '-i').build()
+
+        then:
+        result.task(':yamlRestCompatTest').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyRestApiCompatSpecsTask').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyRestApiCompatTestTask').outcome == TaskOutcome.NO_SOURCE
+    }
+
+    def "yamlRestCompatTest executes and copies api and tests from :bwc:minor"() {
+        given:
+        String api = "wrongversion.json"
+        String test = "wrongversion.yml"
+        setupInternalRestResources(api, test)
+
+        //don't use default distro from here since it requires resolving OS specific project and don't want to create the mock projects
+        System.setProperty("tests.rest.compat.use.default.distro", "false");
```

Comment:
you should be able to get rid of the need for this property by just explicitly configure the test distribution here in the test like adding:
sth like
```
        buildFile << """
        testClusters {
            yamlRestCompatTest.setTestDistribution(TestDistribution.INTEG_TEST)
        }
        """
```

Reply:
this needs to be added after the plugin is applied though otherwise the test cluster doesn't exist yet

Reply:
thanks! fixed in https://github.com/elastic/elasticsearch/pull/64186/commits/249f64f05d572de4c0480ab87fae858ee7902442 , much better then the System property.